snappi_ixnetwork/snappi_api.py

```diff
@@ -301,136 +303,94 @@ def _protocols_exists(self):
+            request_payload = getattr(control_obj, control_choice)
             self._connect()
-            with Timer(self, "Setting Protocol state"):
-                self.ngpf.set_protocol_state(payload)
+            if control_option == "port":
+                if control_choice == "capture":
+                    self.capture.set_capture_state(request_payload)
+                elif control_choice == "link":
+                    self.vport.set_link_state(request_payload)
+            elif control_option == "protocol":
+                if control_choice == "all":
+                    self.ngpf.set_protocol_state(request_payload)
+                elif control_choice == "route":
+                    self.ngpf.set_route_state(request_payload)
+                elif control_choice == "lacp":
+                    self.ngpf.set_device_state(request_payload)
+            elif control_option == "traffic":
+                self.traffic_item.transmit(request_payload)
         except Exception as err:
             raise SnappiIxnException(err)
         return self._request_detail()
-
-    def set_transmit_state(self, payload):
-        """Set the transmit state of flows"""
+    
+    def set_control_action(self, payload):
         try:
-            if isinstance(payload, (type(self._transmit_state), str)) is False:
-                raise TypeError(
-                    "The content must be of type Union[TransmitState, str]"
-                )
-            if isinstance(payload, str) is True:
-                payload = self._transmit_state.deserialize(payload)
-            self._connect()
-            self.traffic_item.transmit(payload)
+            control_option = payload.choice
+            control_obj = getattr(payload, control_option)
+            control_choice = control_obj.get("choice")
+            choice_obj = getattr(control_obj, control_choice)
+            if control_choice == "ipv4":
+                choice = choice_obj.get("choice")
+                request_payload = getattr(choice_obj, choice)
+                if choice == "ping":
+                    res = self.control_action_response()
+                    self._connect()
+                    res.response.protocol.ipv4.ping.responses.deserialize(self.ping.results(control_choice, request_payload))
+            elif control_choice == "ipv6":
+                choice = choice_obj.get("choice")
+                request_payload = getattr(choice_obj, choice)
+                if choice == "ping":
+                    res = self.control_action_response()
+                    self._connect()
+                    res.response.protocol.ipv6.ping.responses.deserialize(self.ping.results(control_choice, request_payload))
+            res.warnings=snappi.Warning()
+            return res
         except Exception as err:
             raise SnappiIxnException(err)
-        return self._request_detail()
+               
+    def set_protocol_state(self, payload):
```

Comment:
This breaks backward compatibility right.
if a old script is using set_protocol_state, now its throws a warning which is right but it wont do any operation right

Reply:
In new model it is deprecated. Is that we need to think of backward compatibility or the script should be inline with new model? Could you please clarify.